### PR TITLE
feat: Left menu review Bookmark space from the left menu (desktop) - MEED-419 - Meeds-io/meeds#16

### PR DESCRIPTION
**Pull request by SaraBoutej**
This Changes allow to bookmark/unbookmark a space from the last visited spaces in the left navigation drawer.

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java [hunk 1]
```diff
@@ -172,6 +199,23 @@ public void setName(String name) {
   public String getDisplayName() {
     return displayName;
   }
-  
-  
+
+  /**
+   * Gets if the space is bookmarked or not.
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]:
```suggestion
   * Check whether the space is favorite or not.
```

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java [hunk 2]
```diff
@@ -172,6 +199,23 @@ public void setName(String name) {
...
+   *
+   * @return
+   */
+  public String getIsFavorite() {
+    return isFavorite;
+  }
+
+  /**
+   * Sets if the space is bookmarked or not.
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]:
```suggestion
   * Sets the checks if is favorite space.
```

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java [hunk 3]
```diff
@@ -16,13 +16,22 @@
  */
 package org.exoplatform.social.service.rest;
 
+import org.exoplatform.container.ExoContainerContext;
+import org.exoplatform.social.core.identity.model.Identity;
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: useless imports!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java [hunk 4]
```diff
@@ -16,13 +16,22 @@
...
 import org.exoplatform.social.core.service.LinkProvider;
 import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.metadata.favorite.FavoriteService;
+import org.exoplatform.social.metadata.favorite.model.Favorite;
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: useless imports!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java [hunk 5]
```diff
@@ -59,19 +68,37 @@ public class SpaceRest {
...
+
+    this.isFavorite = isFavorite;
+  }
+
+  /**
+   * Gets the space id .
+   *
+   * @return
+   *
+   */
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]:
```suggestion
  /**
   * Gets the space id .
   *
   * @return the space id
   */
```

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java [hunk 6]
```diff
@@ -59,19 +68,37 @@ public class SpaceRest {
...
+  public SpaceRest(Space space, String isFavorite) {
+    this.id = space.getId();
     this.url = space.getUrl();
     this.groupId = space.getGroupId();
     this.name = space.getName();
     this.displayName = space.getDisplayName();
     
     this.avatarUrl = space.getAvatarUrl() == null ? LinkProvider.SPACE_DEFAULT_AVATAR_URL : space.getAvatarUrl();
+
+    this.isFavorite = isFavorite;
```
Comment (AzmiTouil) [inline component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: Should be retrieved from space item IMO.

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java (hunk 6 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: We don't have this information in the space entity!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java (hunk 1 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: done!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java (hunk 2 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: Done!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java (hunk 3 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: Removed!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java (hunk 4 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: Removed!

File: component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java (hunk 5 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/service/rest/SpaceRest.java]: fixed!

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**